### ISSUE TITLE
[no-unused-vars] enum members throw errors

### ISSUE THREAD

**Issue by mogafk** [bradzacher/eslint-plugin-typescript#235]
**Repro**

```TS
enum FormFieldIds {
	PHONE = 'phone',
	EMAIL = 'email',
}
interface IFoo {
	fieldName: FormFieldIds,
}
```

**Expected Result**

I expected that code does not cause errors

**Actual Result**

Throw errors:
'PHONE' is assigned a value but never used.
'EMAIL' is assigned a value but never used.

**Additional Info**
You can reproduce behavior on your computed if download repository:
https://github.com/mogafk/reproduce-enum-errors

also i create [pr](https://github.com/bradzacher/eslint-plugin-typescript/pull/236) with fix. But i not sure that it real bug. Sorry if i wrong

**Versions**

"eslint": "^5.10.0",
"eslint-plugin-typescript": "^0.14.0",
"typescript": "^3.1.1",
"typescript-eslint-parser": "^21.0.2",

**Comment by bradzacher** [comment]
A thing to note here is that your parser version is higher than supported by v0.14.0

Though looking at it and testing against master, this is a case we missed.
Good catch!
